python/thrift/appengine-ssl-gateway/appengine/bigtable.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Google App Engine application which connects to Google Cloud Bigtable via
+the HBase Thrift proxy servers."""
+
+# Standard Python libraries.
+import logging
+import os
+
+# Libraries provided by App Engine.
+import jinja2
+import webapp2
+
+# Additional imports we provide ourselves.
+import happybase
+from happybase.hbase import Hbase
+from happybase.hbase import ttypes
+from thrift.transport.TSSLSocket import TSSLSocket
+
+JINJA_ENVIRONMENT = jinja2.Environment(
+    loader=jinja2.FileSystemLoader(os.path.dirname(__file__)),
+    autoescape=True,
+    extensions=['jinja2.ext.autoescape'])
+
+KEY_FILE = 'stunnel.pem'
+
+
+class SecureConnection(happybase.Connection):
+    def _refresh_thrift_client(self):
+        """Refresh the Thrift socket, transport, and client."""
+        socket = TSSLSocket(self.host, self.port, False, KEY_FILE, KEY_FILE,
+            KEY_FILE)
+        if self.timeout is not None:
+            socket.setTimeout(self.timeout)
+        self.transport = self._transport_class(socket)
+        protocol = self._protocol_class(self.transport)
+        self.client = Hbase.Client(protocol)
+
+
+def CreateConnection():
+    # Change this to be your load balancer's IP address.
+    # TODO: add automatic lookup, via Compute APIs? Cache results.
+    return SecureConnection('<Thrift gateway load balancer IP>', 1090)
+
+
+class IndexHandler(webapp2.RequestHandler):
+
+    def get(self):
+        connection = CreateConnection()
+        tables = connection.tables()
+
+        variables = {'tables': tables}
+        template = JINJA_ENVIRONMENT.get_template('web/index.jinja')
+        self.response.write(template.render(variables))
+
+
+class CreateTableHandler(webapp2.RequestHandler):
+    def get(self):
+        connection = CreateConnection()
+        table = 'mytable'
+        status = 'does not exist'
+        try:
+            connection.create_table(
+                'mytable',
+                {'cf1': dict(max_versions=10),
```

Comment:
Exactly, this is just a demo to create/list/delete tables, we don't actually store any data there.